backend/src/middleware/validation/validationMiddleware.js

```diff
@@ -52,29 +52,9 @@ const validatePost = async (resolve, root, args, context, info) => {
 }
 
 const validateUpdatePost = async (resolve, root, args, context, info) => {
-  const { id, categoryIds } = args
-  const session = context.driver.session()
-  const categoryQueryRes = await session.run(
-    `
-    MATCH (post:Post {id: $id})-[:CATEGORIZED]->(category:Category)
-    RETURN category`,
-    { id },
-  )
-  session.close()
-  const [category] = categoryQueryRes.records.map(record => {
-    return record.get('category')
-  })
-
-  if (category) {
-    if (categoryIds && categoryIds.length > 3) {
-      throw new UserInputError(NO_CATEGORIES_ERR_MESSAGE)
-    }
-  } else {
-    if (!Array.isArray(categoryIds) || !categoryIds.length || categoryIds.length > 3) {
-      throw new UserInputError(NO_CATEGORIES_ERR_MESSAGE)
-    }
-  }
-  return resolve(root, args, context, info)
+  const { categoryIds } = args
+  if (typeof categoryIds === 'undefined') return resolve(root, args, context, info)
```

Comment:
what if someone passes `categoryIds but the value is null or  an empty Array`

Reply:
I hope that `[]` gets caught by the validation but I'm going to check `null` (that's equivalent to not supplying categoryIds I would say)

Reply:
Looks alright:
![Screenshot - 2019-09-03T134027 430](https://user-images.githubusercontent.com/2110676/64170348-309fdf00-ce50-11e9-9b9d-f6334b2bb33c.png)